justfile

```diff
@@ -440,7 +451,62 @@ _linkerd-viz-uninit:
 
 
 # TODO linkerd-jaeger-install
-# TODO linkerd-multicluster-install
+
+##
+## linkerd multicluster
+## 
+
+mc-gateway-image := "gcr.io/google_containers/pause:3.2"
+
+linkerd-mc *flags: _k3d-init
+    {{ _linkerd }} mc {{ flags }}
+
+linkerd-mc-install: _linkerd-init linkerd-mc-load && _linkerd-mc-ready
+    {{ _linkerd }} mc install \
+            --set='linkerdVersion={{ linkerd-tag }}' \
+        | {{ _kubectl }} apply -f -
+
+# Wait for all test namespaces to be removed before uninstalling linkerd-multicluster.
+linkerd-mc-uninstall:
+    {{ _linkerd }} mc  uninstall \
+        | {{ _kubectl }} delete -f -
+
+linkerd-mc-build:
+	TAG={{ linkerd-tag }} bin/docker-build-controller
+
+linkerd-mc-load: _linkerd-mc-images _k3d-init
+	{{ _k3d-load }} \
+		{{ controller-image }}:{{ linkerd-tag }} \
+		{{mc-gateway-image }}
+
+_linkerd-mc-images:
+	#!/usr/bin/env bash
+	set -euo pipefail
+	for img in \
+		'{{ mc-gateway-image }}' \
+		'{{ controller-image }}:{{ linkerd-tag }}'
+	do
+		if [ -z $(docker image ls -q "$img") ]; then
+			exec {{ just_executable() }} \
+				linkerd-tag='{{ linkerd-tag }}' \
+				linkerd-mc-build
+		fi
+	done
```

Comment:
The `exec` seems like a problem to me -- once this is run the loop will stop (since `exec` replaces the bash process completely. I think it can just be omitted.

But I don't understand why we'd want to run `linkerd-mc-build` when the pause container isn't present...

Reply:
I see that this is pulled from `_linkerd-images`, but the context is slightly different. I think we can simplify...

Reply:
Ah, yeah makes sense on the exec... my bad, thought for some reason that was necessary.